packages/app-store/routing-forms/pages/route-builder/[...appPages].tsx

```diff
@@ -201,6 +202,126 @@ const buildEventsData = ({
   return { eventOptions, eventTypesMap };
 };
 
+const isValidAttributeIdForWeights = ({
+  attributeIdForWeights,
+  jsonTree,
+}: {
+  attributeIdForWeights: string;
+  jsonTree: JsonTree;
+}) => {
+  if (!attributeIdForWeights || !jsonTree.children1) {
+    return false;
+  }
+
+  return Object.values(jsonTree.children1).some((rule) => {
+    if (rule.type !== "rule" || rule?.properties?.field !== attributeIdForWeights) {
+      return false;
+    }
+
+    const values = rule.properties.value.flat();
+    return values.length === 1 && values.some((value: string) => isDynamicOperandField(value));
+  });
+};
+
+const WeightedAttributesSelector = ({
```

Comment:
It contains all the new weighted attributes logic, making it separately testable and keeping Route component small.